File: Arrays/1. Reverse an Array/reverseArray.cpp
// Reverse an array

// 1. Brute force : traverse the array from right to left and store in a temp array.
// 2. Using start and end pointer

void reverseString(vector<char>& s) {
        int start = 0;
        int end = s.size()-1;
        
        while(start < end){
            swap(s[start], s[end]);
            start++;
            end--;
        }
        return;
}

File: Arrays/1. Reverse an Array/reverseArray.java
public void reverseString(char[] s) {
    int start = 0;
    int end = s.length-1;
    char t;
    while(start < end){
        t = s[start];
        s[start] = s[end];
        s[end] = t;
        
        start++;
        end--;
    }
}